src/services/node-checker.ts

```diff
@@ -0,0 +1,376 @@
+
+    if (relayResponse instanceof Error) {
+      return { success: false, relayResponse, output: 0 }
+    }
+
+    let payload = JSON.parse(relayResponse.payload)
+
+    if (Array.isArray(payload)) {
+      payload = payload[0]
+    }
+
+    const successCheck = comparatorFn(payload, comparator)
+
+    return { relayResponse, success: successCheck, output: payload }
+  }
+
+  /**
+   * Performs a consensus relay on all nodes. Meant to be used to slash nodes that fail any of the checks
+   * @param data payload to send to the blockchain.
+   * @param blockchainID Blockchain to request data from.
+   * @param aat Pocket Authentication token object.
+   * @param path  optional. Blockchain's path to send the request to.
+   * @returns relay response from the blockchain.
+   */
+  async sendConsensusRelay(
+    data: string,
+    blockchainID: string,
+    aat: PocketAAT,
+    path?: string
+  ): Promise<RelayResponse | Error> {
+    return this.sendRelay(
+      data,
+      blockchainID,
+      aat,
+      undefined,
+      path,
+      this.updateConfigurationConsensus(this.configuration),
+      true
+    )
+  }
+
+  /**
+   * Helper function to send a relay.
+   * @param data payload to send to the blockchain.
+   * @param blockchainID Blockchain to request data from.
+   * @param aat Pocket Authentication token object.
+   * @param node optional. Node to check.
+   * @param path optional. Blockchain's path to send the request to.
+   * @param configuration optional. Pocket's configuration object
+   * @param consensusEnabled optional. Enable consensus
+   * @returns relay response.
+   */
+  private async sendRelay(
+    data: string,
+    blockchainID: string,
+    aat: PocketAAT,
+    node?: Node,
+    path?: string,
+    configuration?: Configuration,
+    consensusEnabled?: boolean
+  ): Promise<RelayResponse | Error> {
+    const relayResponse = await this.pocket.sendRelay(
+      data,
+      blockchainID,
+      aat,
+      this.updateConfigurationTimeout(configuration || this.configuration),
+      undefined,
+      HTTPMethod.POST,
+      path,
+      consensusEnabled ? undefined : node,
+      consensusEnabled,
+      undefined
+    )
+
+    if (relayResponse instanceof Error) {
+      return relayResponse
+    } else if (relayResponse instanceof RelayResponse && !checkEnforcementJSON(relayResponse.payload)) {
+      // Unhandled error
+      return new RpcError('0', `Unhandled Error: ${relayResponse.payload}`, undefined, node?.publicKey)
+    }
+
+    return relayResponse as RelayResponse
+  }
+
+  /**
+   * Update the configuration's objects consensus value.
+   * @param pocketConfiguration Pocket's Configuration object.
+   * @returns updated configuration object.
+   */
+  private updateConfigurationConsensus(pocketConfiguration: Configuration): Configuration {
+    return new Configuration(
+      pocketConfiguration.maxDispatchers,
+      pocketConfiguration.maxSessions,
+      5,
+      2000,
+      false,
```

Comment:
I think we could lift these arbitrary values into constants at the top of the file.